src/Microsoft.ApiDesignGuidelines.Analyzers/Core/NonConstantFieldsShouldNotBeVisible.cs

```diff
@@ -32,8 +33,23 @@ public abstract class NonConstantFieldsShouldNotBeVisibleAnalyzer : DiagnosticAn
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
-        { 
-            
+        {
+            analysisContext.RegisterSymbolAction(obj =>
+            {
+                var field = (IFieldSymbol)obj.Symbol;
+
+                if (!field.IsPublic())
+                {
+                    return;
+                }
+
+                if (field.IsConst || (field.IsStatic && field.IsReadOnly))
+                {
+                    return;
+                }
+
+                obj.ReportDiagnostic(field.CreateDiagnostic(Rule));
```

Comment:
So this rule will fire even for non-static fields but the FXCop [implementation](http://index/#UsageRules/NonConstantFieldsShouldNotBeVisible.cs,20) only fires for static fields. @nguerrera @michaelcfanning do you recall why FXCop rule restricts it to static fields?